core/src/main/java/net/sourceforge/jnlp/util/UrlUtils.java

```diff
@@ -341,6 +315,45 @@ public static boolean urlEquals(final URL u1, final URL u2) {
         }
         return false;
     }
+    /**
+     * Checks whether <code>url</code> is relative (or equal) to <code>codebaseUrl</code>.
+     * both urls are processed by sanitizeLastSlash before actual equals. So
+     * protocol://som.url/some/path/ is same as protocol://som.url/some/path. Even
+     * protocol://som.url/some/path\ is same as protocol://som.url/some/path/
+     *
+     * This method returns false in case <code>url</code> contains parent directory notation "..".
+     * See JNLP specification version 9, 3.4: 'A relative URL cannot contain parent directory notations, such as "..". It must denote a file that is stored in a subdirectory of the codebase.'
+     * @param url the url to check
+     * @param codebaseUrl the url to check against
+     * @return true if <code>url</code> is relative to <code>codebaseUrl</code>
+     */
+    public static boolean urlRelativeTo(URL url, URL codebaseUrl) {
+        if (codebaseUrl == url) {
+            return true;
+        }
+        if (codebaseUrl == null || url == null) {
+            return false;
+        }
+        try {
+            URL nu = sanitizeLastSlash(normalizeUrl(url));
+            URL nup = sanitizeLastSlash(normalizeUrl(codebaseUrl));
+            if (!getHostAndPort(nu).equals(getHostAndPort(nup))) {
+                return false;
+            }
+            if (!nu.getProtocol().equals(nup.getProtocol())) {
+                return false;
+            }
+            if (nu.getPath().contains("..")) {
+                return false;
+            }
+            if (nu.getPath().startsWith(nup.getPath())) {
+                return true;
+            }
+        } catch (MalformedURLException | UnsupportedEncodingException | URISyntaxException e) {
+            LOG.error(IcedTeaWebConstants.DEFAULT_ERROR_MESSAGE, e);
```

Comment:
Exception should be rethrown

Reply:
disagree on this. returning false is appropriate in this case.

Reply:
@sclassen but the old implementation throws an exception.

Reply:
In case of MalformedURLException

Reply:
Yes, never the less I think the method should return false. If the URL is really malformed then it will trigger the `MalformedURLException` when ITW will try to download the resource. Which I find a better location to throw the exception.

Reply:
I tend to side with @sclassen on this one, as MalformedURLException can be triggered by sanitizeLastSlash or normalizeUrl and in the current state these methods only throw MalformedURLException  because they use the URL constructor, but in effect it seems that they should never actually throw it as the constructed URL comes from an already valid URL (might be wrong on some edge cases in normalizeUrl method, I'm no expert).
That being said, I'll rethrow the exception if required :)